docs/public/_redirects

```diff
@@ -415,4 +418,4 @@ https://v4.material-ui.com/* https://v4.mui.com/:splat 301!
 /toolpad/* https://mui-toolpad-docs.netlify.app/toolpad/:splat 200
 /r/toolpad-* https://mui-toolpad-docs.netlify.app/r/toolpad-:splat 200
 
-/* https://mui.com/:splat 301!
+/* https://mui.com/:splat 302!
```

Comment:
This line is causing a side-effect - deploy previews from PRs are redirected to `mui.com` 🙃
See https://deploy-preview-36296--material-ui.netlify.app/

I guess the goal was to redirect `next.mui.com/*` => `mui.com/*`, so maybe to this instead?
```suggestion
https://next.mui.com/* https://mui.com/:splat 302!
```

cc @m4theushw

Reply:
It's probably also the reason why `test_e2e_website` fails

Reply:
We actually have this exact redirect, just commented out on L11.
Maybe uncomment it and replace it with 302 while also removing this one? 🤔

Reply:
Let's give it a try